### ISSUE TITLE
Add EXAONE 4.0 and 4.5 support

### ISSUE THREAD

**Issue by DHdroid** [ollama/ollama#16662]
Hi,

We are working on adding support for LG AI Research's [EXAONE 4.0](https://huggingface.co/LGAI-EXAONE/EXAONE-4.0-32B) and [EXAONE 4.5](https://huggingface.co/LGAI-EXAONE/EXAONE-4.5-33B) models to Ollama, and we would like to discuss the expected scope and upstreaming process before opening a PR.

The work is currently available in this branch of our fork:

- https://github.com/DHdroid/ollama/tree/add-exaone4_5

Here is a brief overview of what we have implemented.

## Common

- Added parser and renderer support for EXAONE models.

## GGML / GGUF

- Since Ollama now runs GGUF models through `llama-server`, and EXAONE support already exists in llama.cpp, we did not add a separate Ollama-native GGML model implementation.
- To make `ollama create` from safetensors work, we added safetensors conversion support for EXAONE 4.0 and EXAONE 4.5.

## MLX

- Added MLX model implementations for EXAONE 4.0 and EXAONE 4.5.
- Added an EAGLE/NEXTN-style MTP draft model for EXAONE 4.5.
- Since this style of MTP was not already implemented in Ollama's MLX runner, we added a local implementation. We also noticed the new works on MTP & cache have been merged, so we expect the MTP part may need to be rebased or adjusted on top of that direction before it is ready for review. The main pieces of our current implementation are:

  - Added a draft model that maintains its own KV cache. During speculative decoding, the draft model uses temporary draft cache state while proposing tokens, then the persistent draft cache is updated from the target model's verified hidden states after token acceptance.
  - Added draft cache snapshots to the prefix trie so prefix caching can also restore MTP draft KV state. Without this, the draft KV cache has to be rebuilt from target-model hidden states after a prompt-cache hit.
  - Updated the speculative decoding loop:
    - The existing MTP path uses a three-step loop: single-token target decode, draft generation, and target validation. See [`runGreedyMTPDecode`](https://github.com/ollama/ollama/blob/1abd56b6e68f266154257b49c55aa5aa9fb598c3/x/mlxrunner/mtp.go#L193).
    - For EAGLE/NEXTN-style MTP, the single-token decode and validation steps overlap. The validation forward already produces the next target token, so doing another single-token decode immediately afterward is redundant.
    - We convert this loop into a two-step loop: draft from the previous validation state, then validate. The target token produced by validation is carried as the next pending token.
    - This is implemented in [`runGreedyEagleMTPDecode`](https://github.com/DHdroid/ollama/blob/7188819183ef800846f02db4cac80f48a35c42f7/x/mlxrunner/mtp.go#L488). The existing `runGreedyMTPDecode` path remains same as the previous MTP flow.

- We have not added multimodal EXAONE 4.5 support yet. The current implementation is text-only. We plan to look at multimodal support once the expected MLX multimodal/projector path is clearer.

Example Modelfiles:

```modelfile
# EXAONE 4.0
FROM /path/to/your/exaone-4.0-safetensors

RENDERER exaone4
PARSER exaone4

PARAMETER temperature 0.6
PARAMETER top_p 0.95
PARAMETER presence_penalty 1.5
PARAMETER repeat_penalty 1.0
PARAMETER repeat_last_n 0
```

```modelfile
# EXAONE 4.5
FROM /path/to/your/exaone-4.5-safetensors

RENDERER exaone4_5
PARSER exaone4_5

PARAMETER temperature 1.0
PARAMETER top_p 0.95
PARAMETER presence_penalty 1.5
PARAMETER repeat_penalty 1.0
PARAMETER repeat_last_n 0
```

For EXAONE 4.5, we observed poor generation quality with the default repetition penalty in our local MLX tests, likely due to unusually large logits. Setting `repeat_penalty` to `1.0` avoided that issue in our testing.

## Questions

What would be the preferred next step to add EXAONE 4.0 and EXAONE 4.5 as officially supported models in Ollama?

In particular, would you prefer that we first open a PR for EXAONE 4.0/4.5 text-only support, and then follow up with EXAONE 4.5 MTP support after the basic model support lands?

Any guidance on the preferred scope and review path would be very helpful. We would be happy to adjust the branch, split the work if needed, or provide additional testing details before opening a PR.

Thank you for your time!

**Comment by DHdroid** [comment]
@rick-github Hi, just checking in on this. Is there anything I can do to help move this forward? I made a [new branch](https://github.com/DHdroid/ollama/tree/add-exaone4_5-model) which contains the text-only support (w/o MTP), which might be much easier to start with. Would it be ok to open a new PR with this branch?

I’d be happy to help with anything else needed to get our models supported in Ollama!

**Comment by rick-github** [comment]
> [@rick-github](https://github.com/rick-github) Hi, just checking in on this. Is there anything I can do to help move this forward? I made a [new branch](https://github.com/DHdroid/ollama/tree/add-exaone4_5-model) which contains the text-only support (w/o MTP), which might be much easier to start with. Would it be ok to open a new PR with this branch?
> 
> I’d be happy to help with anything else needed to get our models supported in Ollama!

@dhiltgen @jmorganca

**Comment by dhiltgen** [comment]
This model architecture is now supported. For example `ollama run hf.co/LGAI-EXAONE/EXAONE-4.0-1.2B-GGUF:Q4_K_M`
